File: packages/react-native-performance/ios/ReactNativePerformance/PerformanceMarkerManager.m
#import <Foundation/Foundation.h>
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(PerformanceMarkerManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(renderPassName, NSString)
RCT_EXPORT_VIEW_PROPERTY(interactive, BOOL)
RCT_EXPORT_VIEW_PROPERTY(reportOnDraw, BOOL)
RCT_EXPORT_VIEW_PROPERTY(destinationScreen, NSString)
RCT_EXPORT_VIEW_PROPERTY(componentInstanceId, NSString)
RCT_EXPORT_VIEW_PROPERTY(onRenderComplete, RCTDirectEventBlock)

@end

File: packages/react-native-performance/ios/ReactNativePerformance/PerformanceMarkerManager.swift
import Foundation

@objc(PerformanceMarkerManager)
class PerformanceMarkerManager: RCTViewManager {
    override func view() -> UIView! {
        return PerformanceMarker()
    }

    override static func requiresMainQueueSetup() -> Bool {
        return true
    }
}
